JSON Web Tokens Implemented

juliabreeden opened the pull request:
## Description

Creates a JSON web token in the log in and signup routes and an authenticate token middleware placed in every authorized route. 

## Asana Task



## Checklist

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description** and **Asana Task** sections above.
- [ ] I added or updated Jest unit tests for any changes to components, server-side controllers, etc.
- [ ] I added or updated Cypress end-to-end tests for any changes to page, the application process, and other major front-end flows.

juliabreeden committed:
JWT and logout route added

juliabreeden committed:
Merge branch 'dev' of https://github.com/CampfireConnect/adventure-connect into JB_JWT

juliabreeden committed:
JWTs ready and signup error bug removed, chandler to complete signup error handling

cchrty commented on server/controllers/userControllers.js:
```diff
@@ -28,8 +30,19 @@ userController.verifyLogin = async (req, res, next) => {
+      //add something that adds the JWT here
+      const currentUser = { user: user._id};
+      const accessToken = jwt.sign(currentUser, process.env.ACCESS_TOKEN_SECRET, { expiresIn: '1800s'});
+      res.locals.accessToken = accessToken;
```
why does this also get stored in res.locals? (just curious!)

cchrty committed:
Update userControllers.js

removed commented out code > moved all cloud storage code into imageController file
